PhysicsTools/NanoAOD/plugins/P2GTTriggerResultsConverter.cc

```diff
@@ -0,0 +1,91 @@
+
+#include <string>
+
+using namespace l1t;
+
+class P2GTTriggerResultsConverter : public edm::stream::EDProducer<> {
+public:
+  explicit P2GTTriggerResultsConverter(const edm::ParameterSet&);
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  enum DecisionType { beforeBxMaskAndPrescale, beforePrescale, final };
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  void beginRun(const edm::Run&, const edm::EventSetup&) override;
+
+  const edm::EDGetTokenT<P2GTAlgoBlockMap> algoBlockToken_;
+  const std::string prefix_;
+  const std::string decisionName_;
+  const DecisionType decisionEnum_;
+
+  std::vector<std::string> algoNames_;
+};
+
+P2GTTriggerResultsConverter::P2GTTriggerResultsConverter(const edm::ParameterSet& params)
+    : algoBlockToken_(consumes<P2GTAlgoBlockMap>(params.getParameter<edm::InputTag>("src"))),
+      prefix_(params.getParameter<std::string>("prefix")),
+      decisionName_(params.getParameter<std::string>("decision")),
+      decisionEnum_(decisionName_ == "beforeBxMaskAndPrescale" ? beforeBxMaskAndPrescale
+                    : decisionName_ == "beforePrescale"        ? beforePrescale
+                                                               : final) {
+  produces<edm::TriggerResults>();
+}
+
+void P2GTTriggerResultsConverter::beginRun(const edm::Run&, const edm::EventSetup&) { algoNames_.clear(); }
+
+void P2GTTriggerResultsConverter::produce(edm::Event& event, const edm::EventSetup&) {
+  const P2GTAlgoBlockMap& algoBlockMap = event.get(algoBlockToken_);
+
+  edm::HLTGlobalStatus gtDecisions(algoBlockMap.size());
+
+  bool fillAlgoNames = false;
+
+  if (algoNames_.size() == 0) {
+    algoNames_ = std::vector<std::string>(algoBlockMap.size());
+    fillAlgoNames = true;
+  }
+
+  std::size_t algoIdx = 0;
+  for (const auto& [algoName, algoBlock] : algoBlockMap) {
+    bool decision = decisionEnum_ == beforeBxMaskAndPrescale ? algoBlock.decisionBeforeBxMaskAndPrescale()
+                    : decisionEnum_ == beforePrescale        ? algoBlock.decisionBeforePrescale()
+                                                             : algoBlock.decisionFinal();
+
+    gtDecisions[algoIdx] = edm::HLTPathStatus(decision ? edm::hlt::Pass : edm::hlt::Fail);
+    if (fillAlgoNames) {
+      algoNames_[algoIdx] = prefix_ + algoName + "_" + decisionName_;
+    }
+    algoIdx++;
+  }
+
+  event.put(std::make_unique<edm::TriggerResults>(gtDecisions, algoNames_));
+}
+
+void P2GTTriggerResultsConverter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("src");
+  desc.add<std::string>("prefix", "L1_");
+  desc.ifValue(edm::ParameterDescription<std::string>("decision", "final", true),
+               edm::allowedValues<std::string>("beforeBxMaskAndPrescale", "beforePrescale", "final"));
+  descriptions.addWithDefaultLabel(desc);
+}
+
+DEFINE_FWK_MODULE(P2GTTriggerResultsConverter);
```

Comment:
where does this module get used ? should this go in one of the custom L1 nano ?

Reply:
@artlbv wants to use this to get the trigger decisions accessible by their trigger name into nano, analogously to [`L1TriggerResultsConverter`](https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/plugins/L1TriggerResultsConverter.cc). Maybe he could comment on whether this should be in one of those configs.

Reply:
Indeed, we use that in the Phase-2 L1 nano which for now is outside of CMSSW: https://github.com/cms-l1-dpg/Phase2-L1Nano/blob/dev_wGT_trigResults/python/l1tPh2GTtables_cff.py#L14-L19

Reply:
Ah, @vlimant do you mean we should put that e.g. under `DPGAnalysis/L1TNanoAOD/plugins` as for the phase-1 DPG nano plugins: https://github.com/cms-sw/cmssw/tree/master/DPGAnalysis/L1TNanoAOD/plugins ?
In principle, this is a "custom" nano for the moment, so it might make sense (outside XPOG). 

OTOH, for phase-2 this will be the new default way of saving the L1 GT decision bits into nano. 
So if we put it into the DPG now, we will have to move it to the central nano (within XPOG) at some point.

What would be the best way @vlimant @aloeliger @epalencia ?

Reply:
the move should be done by @HaarigerHarald - can you do that?

On the customisation: this PR is for the Global Trigger decisions, thus it should modify this part of the central nano which does not make sense for Phase-1: https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/python/triggerObjects_cff.py#L416-L421
But the `L1/L1Full` are only for the L1 objects, not the GT. 

I have the impression that there is still a lot of phase-1 content in the phase-2 nano, therefore I never used the "central" nano for the phase-2 studies.

Reply:
@artlbv @vlimant is https://github.com/cms-sw/cmssw/pull/44995/commits/ac28e88aacffb45ce884bb331d279a165ae75f7c what you had in mind?

Reply:
please locate this plugin  in `DPGAnalysis/L1TNanoAOD`

Reply:
@vlimant Done